types/index.d.ts

```diff
@@ -2058,7 +2058,7 @@ declare module 'mongoose' {
    * { age: 30 }
    * ```
    */
-  export type UpdateQuery<T> = _UpdateQuery<_UpdateQueryDef<T>> & AnyObject;
+  export type UpdateQuery<T> = Partial<T> & _UpdateQuery<_UpdateQueryDef<T>> & AnyObject;
```

Comment:
I think the reason is the Union Type with AnyObject resulting in a Type Widening. I would usually try with Omit<AnyObject, keyof T & _UpdateQuery<_UpdateQueryDef<T>>> instead of just using AnyObjecy.

```suggestion
  export type UpdateQuery<T> = Partial<T> & _UpdateQuery<_UpdateQueryDef<T>> & Omit<AnyObject, keyof T & _UpdateQuery<_UpdateQueryDef<T>>>;
```

Didn't test it, just wrote it with my mobile.

Also look into the populate typing PR. @mohammad0-0ahmad used a different notation to achieve the correct typings

Reply:
Or maybe the following, if update query should accept any type:
@vkarpov15  what do u think about that ?
```
export type UpdateQuery<T> = _UpdateQuery<_UpdateQueryDef<T>> & (T | AnyObject);
```

Reply:
@mohammad0-0ahmad that approach works for me, as long as it is `& (Partial<T> | AnyObject)`